components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt

```diff
@@ -84,6 +84,7 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
     @VisibleForTesting
     internal var password by SafeArgString(KEY_LOGIN_PASSWORD)
 
+    private var loginValid = false
```

Comment:
Unfortunately, storing state in a field isn't safe in a fragment, as the fragment can be re-created by the system at any time (e.g. when the user rotates the device), which would result in this state being lost. You could use `safeArguments` (see other params), **or** make sure `loginValid` is set again when the fragment is re-created, based on `password`. That's what I would do. `bindPassword` seems to already be called when this happens and `password` is already restored from `arguments`. So we could set `loginValid` right there in `bindPassword` , or make sure the text change listener runs in this case (`setText` is currently called before the listener is attached).

A second issue here is that the read and write of this field occur on different thread and there are not visibility guarantees. If you keep using a `boolean` (using the second approach above), you can mark it as `@Volatile` which is the easiest solution here.

Reply:
(There's also a pre-existing problem with `validateStateUpdate` being read/written on different threads and it's not thread-safe either. The consequence here is that a job may not get cancelled correctly. This is unrelated to your changes though, just noticed while taking a look at this class for the first time. We can file a follow-up.)

Reply:
I went with the least amount of changes here and called `setText()` after `addTextChangedListener()`.  I feel like a refactor could be good, but the follow-up would be a better place to do that.